.github/workflows/terraformPlan.yml

```diff
@@ -0,0 +1,131 @@
+name: 'Terraform Plan'
+env:
+  terraform_version: "0.12.24"
+on:
+  pull_request:
+    paths:
+      - '!hack/**'
+      - '!.github/**'
+      - '.github/workflows/terraformPlan.yml'
+      - 'environments/*/terraform/**'
+      - 'templates/terraform/**'
+jobs:
+  terraform_plan:
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - environment: dev
+          vault_env: dev
+          vault_role_id_secret: MONSTER_AUTH_ROLE_ID
+          vault_secret_id_secret: MONSTER_AUTH_ROLE_SECRET
+        - environment: hca
+          vault_env: dev
+          vault_role_id_secret: MONSTER_AUTH_ROLE_ID
+          vault_secret_id_secret: MONSTER_AUTH_ROLE_SECRET
+        - environment: hca-prod
+          vault_env: prod
+          vault_role_id_secret: MONSTER_PROD_AUTH_ROLE_ID
+          vault_secret_id_secret: MONSTER_PROD_AUTH_ROLE_SECRET
+        - environment: prod
+          vault_env: prod
+          vault_role_id_secret: MONSTER_PROD_AUTH_ROLE_ID
+          vault_secret_id_secret: MONSTER_PROD_AUTH_ROLE_SECRET
+        - environment: v2f-prod
+          vault_env: prod
+          vault_role_id_secret: MONSTER_PROD_AUTH_ROLE_ID
+          vault_secret_id_secret: MONSTER_PROD_AUTH_ROLE_SECRET
+    env:
+      terraform_directory: "${{ github.workspace }}/environments/${{ matrix.environment }}/terraform"
+      VAULT_ADDR: ${{ secrets.VAULT_ADDR }}
+      GOOGLE_APPLICATION_CREDENTIALS: "${{ github.workspace }}/environments/gcs_sa_key.json"
+      AWS_REGION: us-east-1
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+    if: "!contains( github.event.pull_request.labels.*.name, 'skip-ci')"
+    name: "${{ matrix.environment }} Terraform Plan"
+    runs-on: ubuntu-latest
+    steps:
+      - name: "${{ matrix.environment }} Checkout"
+        uses: actions/checkout@v2
+
+      - name: install terraform
+        uses: hashicorp/setup-terraform@v1.2.1
+        with:
+          terraform_version: ${{ env.terraform_version }}
+
+      - name: "${{ matrix.environment }} set VAULT_TOKEN"
+        id: token
+        run: |
+          VAULT_TOKEN=$(curl \
+            --request POST \
+            --data '{"role_id":"'"${ROLE_ID}"'","secret_id":"'"${SECRET_ID}"'"}' \
+            ${VAULT_ADDR}/v1/auth/approle/login | jq -r .auth.client_token)
+          echo ::add-mask::${VAULT_TOKEN}
```

Comment:
👍 til about ::add-mask in github workflows

Reply:
yeah the ability to dynamically mark things as sensitive is really great